File: Sum of Two Values/SumOfTwoValues.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Scanner;

public class SumOfTwoValues {
    static Scanner input = new Scanner(System.in);

    public static void main(String[] args) {
        int n = input.nextInt();
        int x = input.nextInt();
        int[] tem = new int[n];
        HashMap<Integer, Integer> arr = new HashMap<Integer, Integer>();
        for (int i = 0; i < n; i++) {
            tem[i] = input.nextInt();
            arr.put(tem[i], i + 1);
        }
        for (int i = 0; i < n; i++) {
            int temp = x - tem[i];
            if (arr.containsValue(temp) && arr.get(temp) != i + 1) {
                System.out.println((i + 1) + " " + arr.get(temp));
                System.exit(0);
            }
        }
        System.out.println("IMPOSSIBLE");
    }
}

File: Sum of Two Values/SumOfTwoValues.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n, x, temp, tem;
    scanf("%d %d", &n, &x);
    int arr[n];
    map<int, int> m;
    for(int i=0; i<n; i++){
        scanf("%d", &arr[i]);
        m[arr[i]] = i+1;
    }
    for(int i=0; i<n;i++){
        temp = x - arr[i];
        if(m.count(temp) && m[temp] != i+1){
            cout << i+1 << " " << m[temp];
            return 0;
        }
    }
    cout << "IMPOSSIBLE";
    return 0;
}